sensor_test.py

```diff
@@ -0,0 +1,240 @@
+
+
+def test_update_sensor_firmware(get_sensor_info, update_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+    print("Get original sensor firmware version")
+    sensor_info = get_sensor_info()
+    original_firmware_version = sensor_info.get("firmware_version")
+
+    print("Request firmware update")
+    update_firmware()
+    wait(
+        func=get_sensor_info,
+        condition=lambda x: isinstance(x, dict),
+        tries=5,
+        timeout=3,
+    )
+
+    print("Get current sensor firmware version")
+    updated_sensor_info = get_sensor_info()
+    current_firmware_version = updated_sensor_info.get("firmware_version")
+
+    print("Validate that current firmware version is +1 to original firmware version")
+    assert (
+        current_firmware_version == original_firmware_version + 1
+    ), "Sensor firmware was not updated"
+
+    print("Repeat steps 1-4 until sensor is at max_firmware_version - 1")
+    max_firmware_version = 15
+    while True:
+        update_firmware()
+        wait(
+            func=get_sensor_info,
+            condition=lambda x: isinstance(x, dict),
+            tries=5,
+            timeout=3,
+        )
+        updated_sensor_info = get_sensor_info()
+        current_firmware_version = updated_sensor_info.get("firmware_version")
+        if current_firmware_version == max_firmware_version - 1:
+            break
+
+    print("Update sensor to max firmware version")
+    while True:
+        update_firmware()
+        wait(
+            func=get_sensor_info,
+            condition=lambda x: isinstance(x, dict),
+            tries=5,
+            timeout=3,
+        )
+        updated_sensor_info = get_sensor_info()
+        current_firmware_version = updated_sensor_info.get("firmware_version")
+        if current_firmware_version == max_firmware_version:
+            break
+
+    print("Validate that sensor is at max firmware version")
+    sensor_info = get_sensor_info()
+    current_firmware_version = sensor_info.get("firmware_version")
+    assert (
+        current_firmware_version == max_firmware_version
+    ), "Sensor firmware version is not at max possible"
+
+    print("Request another firmware update")
+    sensor_info = get_sensor_info()
+    firmware_version_before_update = sensor_info.get("firmware_version")
+
+    update_sensor_firmware = update_firmware()
```

Comment:
This variable name is a verb name ("update") but it's better to have them as noun names and leave verbs for functions. Something like `sensor_update_response` would be more appropriate.